hack/make/test-install-script

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# This script is used for testing install.sh and that it works for
+# each of component of our apt and yum repos
+set -e
+
+test_install_script(){
+	# these are equivalent to main, testing, experimental components
+	# in the repos, but its the url that will do the conversion
+	components=( get test experimental )
+
+	for component in "${components[@]}"; do
+		# change url to specific component for testing
+		local test_url=https://${component}.docker.com
+		local script=/tmp/install.sh
+		sed "s,url='https://get.docker.com/',url='${test_url}/'," hack/install.sh > ${script}
+
+		chmod +x ${script}
+
+		# test for each Dockerfile in contrib/builder
+		for dir in contrib/builder/*/*/; do
+			version="$(basename "$dir")"
+			distro="${version%-*}"
+			suite="${version##*-}"
+			from="${distro}:${suite}"
```

Comment:
Do you think this would be simpler as something like `from="$(awk 'toupper($1) == "FROM" { print $2; exit }' "$dir/Dockerfile")"` instead of trying to parse the directory name?